### DISCUSSION TITLE
Easier chaining of helpers that require re-render

### DISCUSSION THREAD

**Discussion by wdfinch** [react-hook-form/react-hook-form#3973] [Ideas]
Right now helpers like `move` or `append` trigger a rerender. Therefore one must complete (trigger the rerender) before another can be called. This can get tedious and messy if multiple helpers need to be called one after the other. 

I propose adding a helper that will chain these together. I see this working by changing the internal state of the form synchronously based on the method and then re-rendering the component once at the end. This not only leads to cleaner code but also better performance.

What do you think?

**Comment by bluebill1049** [comment]
i think this is possible if we introduce another option.

```
append({ test: 'test' }, { shouldRender: false });
prepend({ test: 'test' }, { shouldRender: false });
prepend({ test: 'test' });
```

**Reply by wdfinch** [reply]
Ya that works. However the user must know to have it render on the last one and always included that option on subsequent ones. Might be a little safer and less repetitive￼ to use a method that does the behavior by default.
